### PR TITLE
Create a Contrib Deployment Target for Cloud Foundry

### PR DESCRIPTION
**Pull request by PeteLevineA**
## Hey, I just made a Pull Request!

Create a Contrib Deployment Target for Cloud Foundry

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

Hey Team!
The team that maintains the Cloud Foundry CLI and other various projects was hoping to get in a document on how to deploy backstage to open source cloud foundry. https://www.cloudfoundry.org/ I hope I have followed the other documents examples okay enough. I'm happy to refer more to documentation if you feel some of the code is too verbose etc. Thank you!

### CODE REVIEW

**Comment by aramissennyeydd** [comment]
👋 Adding a link to https://github.com/backstage/backstage/pull/27151 for more context on how this ended up in `/contrib`.

File: contrib/docs/tutorials/cloudfoundry-deployment.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import {
+  authProvidersExtensionPoint,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import { oidcAuthenticator } from '@backstage/plugin-auth-backend-module-oidc-provider';
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { NotFoundError } from '@backstage/errors';
+
+const uaaAuthProviderModule = createBackendModule({
```
Comment (Parsifal-M) [inline contrib/docs/tutorials/cloudfoundry-deployment.md:194]: Should this maybe be `export const uaaAuthProviderModule`? then remove the `export default uaaAuthProviderModule;`?

Maybe a silly NIT but its usually the pattern I notice 😅

**Review by Parsifal-M — APPROVED** [review]
Cool! I think it looks good 👍 I can't comment on the actual instructions, I assume they are correct 😅

File: contrib/docs/tutorials/cloudfoundry-deployment.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+backend.add(import('./oidcProvider'));
+...
+```
+
+#### Configuration
+
+In UAA, create a new OAuth client with scopes that give Backstage the following permissions:
+
+- Ability for the UAA client to issue refresh tokens
+- Obtain a user's openid, profile, and email
```
Comment (camilaibs) [inline contrib/docs/tutorials/cloudfoundry-deployment.md:270]:
```suggestion
- Obtain a user's `openid`, profile, and email
```

File: contrib/docs/tutorials/cloudfoundry-deployment.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+```
+
+To use Yarn v1, create a new `.yarnrc` file in the build directory with the following content:
+
+```
+yarn-offline-mirror "./npm-packages-offline-cache"
+yarn-offline-mirror-pruning true
+```
+
+In the build directory, untar the backstage packages and run `yarn install` to install npm dependencies into the yarn offline mirror directory:
```
[the remark this answers is not in the document]
Comment (PeteLevineA) [reply contrib/docs/tutorials/cloudfoundry-deployment.md:66]: Appreciate the fixes!

**Pull request closed without merging**